pkg/services/librarypanels/database.go

```diff
@@ -62,3 +62,54 @@ func (lps *LibraryPanelService) deleteLibraryPanel(c *models.ReqContext, panelID
 
 	return err
 }
+
+// getLibraryPanels gets Library Panels based on statement and args
+func getLibraryPanels(session *sqlstore.DBSession, statement interface{}, args ...interface{}) ([]LibraryPanel, error) {
+	libraryPanels := make([]LibraryPanel, 0)
+
+	err := session.SQL(statement, args...).Find(&libraryPanels)
+
+	return libraryPanels, err
+}
+
+// getLibraryPanel gets a Library Panel
+func (lps *LibraryPanelService) getLibraryPanel(c *models.ReqContext, panelID int64) (LibraryPanel, error) {
+	orgID := c.SignedInUser.OrgId
+	libraryPanel := LibraryPanel{}
+
+	err := lps.SQLStore.WithDbSession(context.Background(), func(session *sqlstore.DBSession) error {
+		libraryPanels, err := getLibraryPanels(session, "SELECT * FROM library_panel WHERE id=? and org_id=?", panelID, orgID)
+
+		if err != nil {
+			return err
+		} else if len(libraryPanels) != 1 {
+			return errLibraryPanelNotFound
```

Comment:
Shouldn't this be treated as a bug instead?? It should be logged as an error and returned as 500 to the client, or maybe even ` panic` would be warranted since it would be a hard error on our side.

Reply:
why would it be a bug? the panel is just not found with the specified id, would be a simple 404 in the API

Reply:
@torkelo Sorry I didn't think about that case, I thought about the case where `len(libraryPanels) > 1`. I always take that case into account when I write these queries.

Reply:
That doesn't seem so YAGNI to me @aknuds1 ;) But sure I can add that as well in my next commit.

Reply:
Not YAGNI but added this as suggested.

Reply:
That's the first time I hear of any connection between YAGNI and handling possible error cases. Handling possible error cases is only good code hygiene.